### DISCUSSION TITLE
Question about sumBioMass

### DISCUSSION THREAD

**Discussion by JBiaggi** [SysBioChalmers/yeast-GEM#362] [Q&A] [answered]
Hello! I was inspecting the sumBioMass (https://github.com/SysBioChalmers/yeast-GEM/blob/main/code/otherChanges/sumBioMass.m) function and saw that you substract 18 from the molecular weight of most biomass components.

`96                    MW = data.MWs(pos)-18;`

Could you explain why do you do that? I suppose that 18 is the MW of water, and I saw here #159 that you had a problem with water included in polymers, but I didn't get why you do it for the other components.

**Comment by edkerk** [comment, marked as answer]
Considering the excluding if-statements in line 85 and line 93, line 96 is applied for protein; carbohydrate; RNA and DNA. For all of these macromolecules, their polymerization releases a H2O molecule.

```
85    if strcmp(compType,'L')
86        F = -sum(model.S(isSub,rxnPos));   %g/gDW
87    else
88        F = 0;
89        %Add up all components:
90        for i = 1:length(model.mets)
91            pos = strcmp(data.mets,model.mets{i});
92            if isSub(i) && sum(pos) == 1
93                if strcmp(compType,'I') || strcmp(compType,'F')
94                    MW = data.MWs(pos);
95                else
96                    MW = data.MWs(pos)-18;
```
